src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,38 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE_USER = 18;
+    private static final int MINIMUM_LENGTH_PASSWORD = 6;
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLogin(user.getLogin());
+        checkPassword(user.getPassword());
+        if (user.getAge() < MIN_AGE_USER) {
+            throw new RegistrationUserException("User age less than " + MIN_AGE_USER);
```

Comment:
In case of invalid data you should throw RuntimeException.

Reply:
I know this deviation from the assignment, but I think it is more correct.